src/test/script_tests.cpp

```diff
@@ -931,16 +931,34 @@ BOOST_AUTO_TEST_CASE(script_json_test)
     // amount (nValue) to use in the crediting tx
     UniValue tests = read_json(std::string(json_tests::script_tests, json_tests::script_tests + sizeof(json_tests::script_tests)));
 
+    const KeyData keys;
     for (unsigned int idx = 0; idx < tests.size(); idx++) {
         const UniValue& test = tests[idx];
         std::string strTest = test.write();
         CScriptWitness witness;
+        TaprootBuilder taprootBuilder;
         CAmount nValue = 0;
         unsigned int pos = 0;
         if (test.size() > 0 && test[pos].isArray()) {
             unsigned int i=0;
             for (i = 0; i < test[pos].size()-1; i++) {
-                witness.stack.push_back(ParseHex(test[pos][i].get_str()));
+                auto element = test[pos][i].get_str();
+                // We use #SCRIPT# to flag a non-hex script that we can read using ParseScript
+                // Taproot script must be third from the last element in witness stack
+                std::string scriptFlag = std::string("#SCRIPT#");
+                if (element.find(scriptFlag) == 0) {
+                    CScript script = ParseScript(element.substr(scriptFlag.size()));
+                    witness.stack.push_back(ToByteVector(script));
+                } else if (test[pos].size() >= 3 && i == test[pos].size()-2 && strcmp(element.c_str(), "#CONTROLBLOCK#") == 0) {
```

Comment:
Do you really need to check the position? Should be enough to just look for the tag.

Reply:
@EthanHeilman you would be better suited to answer this question

Reply:
@halseth Sorry I missed you question.

I was trying to avoid the case where say the witness stack is empty because someone wrote a test wrong and then the test breaks in a non-obvious way that confuses the person writing the test. Thinking about this more, having the "#CONTROLBLOCK#" flag not trigger in due to stack position is more confusing and is less flexible.

See PR to address this here: https://github.com/0xBEEFCAF3/bitcoin/pull/9/files

Reply:
> See PR to address this here: https://github.com/0xBEEFCAF3/bitcoin/pull/9/files

ACK 👍

Reply:
This if merged and fixed now. Thanks for the help!

Reply:
Am I confused or is that merge not a part of this PR?

Reply:
Thanks for catching this. This got dropped during my squashing